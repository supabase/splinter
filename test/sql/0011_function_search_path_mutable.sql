begin;
  set local search_path = '';

  -- 0 issues
  select * from lint."0011_function_search_path_mutable";

  create or replace function public.abc()
    returns int
	language sql
  as $$
	select 1;
  $$;

  -- 1 issue
  select * from lint."0011_function_search_path_mutable";

  -- Replace function and set search_path
  create or replace function public.abc()
    returns int
	set search_path=''
	language sql
  as $$
	select 1;
  $$;

  -- 1 issue
  select * from lint."0011_function_search_path_mutable";

rollback;
