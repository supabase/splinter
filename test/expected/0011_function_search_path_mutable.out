begin;
  -- 0 issues
  select * from lint."0011_function_search_path_mutable";
 name | level | facing | description | detail | remediation | metadata | cache_key 
------+-------+--------+-------------+--------+-------------+----------+-----------
(0 rows)

  create or replace function abc()
    returns int
	language sql
  as $$
	select 1;
  $$;
  -- 1 issue
  select * from lint."0011_function_search_path_mutable";
             name             | level |  facing  |                                                  description                                                   |                         detail                         |                                           remediation                                            |                        metadata                         |                                cache_key                                 
------------------------------+-------+----------+----------------------------------------------------------------------------------------------------------------+--------------------------------------------------------+--------------------------------------------------------------------------------------------------+---------------------------------------------------------+--------------------------------------------------------------------------
 function_search_path_mutable | WARN  | EXTERNAL | Detects functions with a mutable search_path parameter which could fail to execute sucessfully for some roles. | Function \`public.abc\` has a role mutable search_path | https://supabase.com/docs/guides/database/database-linter?lint=0011_function_search_path_mutable | {"name": "abc", "type": "function", "schema": "public"} | function_search_path_mutable_public_abc_d859abc2e564f01aec8fe94638390925
(1 row)

  -- Replace function and set search_path
  create or replace function abc()
    returns int
	set search_path=''
	language sql
  as $$
	select 1;
  $$;
  -- 1 issue
  select * from lint."0011_function_search_path_mutable";
 name | level | facing | description | detail | remediation | metadata | cache_key 
------+-------+--------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
